### PR TITLE
rename KeeperRegistry to KeeperRegistry1_2

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

File: contracts/src/v0.8/KeeperRegistry1_2.sol [hunk 1]
```diff
@@ -11,16 +11,16 @@ import "./interfaces/TypeAndVersionInterface.sol";
 import "./interfaces/AggregatorV3Interface.sol";
 import "./interfaces/LinkTokenInterface.sol";
 import "./interfaces/KeeperCompatibleInterface.sol";
-import "./interfaces/KeeperRegistryInterface.sol";
```
Comment (FelixFan1992) [inline contracts/src/v0.8/KeeperRegistry1_2.sol:14]: I think I still need to update the interface names in these files? I only updated the file names but in the future we will add new interfaces here so there may be naming conflicts?

Comment (infiloop2) [reply contracts/src/v0.8/KeeperRegistry1_2.sol:14]: I think we only need one of two things
1. Either have different interface names and struct names within the same file
2. Or have different files (your current implementation) with same struct names

I think 2 is more cleaner and is what you already have

File: contracts/src/v0.8/KeeperRegistry1_2.sol [hunk 2]
```diff
@@ -11,16 +11,16 @@ import "./interfaces/TypeAndVersionInterface.sol";
...
-import "./interfaces/MigratableKeeperRegistryInterface.sol";
-import "./interfaces/UpkeepTranscoderInterface.sol";
+import "./interfaces/KeeperRegistryInterface1_2.sol";
+import "./interfaces/MigratableKeeperRegistryInterface1_2.sol";
+import "./interfaces/UpkeepTranscoderInterface1_2.sol";
```
Comment (infiloop2) [inline contracts/src/v0.8/KeeperRegistry1_2.sol:16]: I think for these two we can keep the same for now? Don't think we envision features changing these interfaces?

File: contracts/src/v0.8/dev/CanaryUpkeep.sol [hunk 3]
```diff
@@ -2,7 +2,7 @@
 pragma solidity 0.8.6;
 
 import "../interfaces/KeeperCompatibleInterface.sol";
-import "../interfaces/KeeperRegistryInterface.sol";
+import "../interfaces/KeeperRegistryInterface1_2.sol";
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/CanaryUpkeep.sol:5]: Should we rename CanaryUpkeep to CanaryUpkeep1_2 ?

File: contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.6;
 
-import {KeeperRegistryExecutableInterface} from "../interfaces/KeeperRegistryInterface.sol";
+import {KeeperRegistryExecutableInterface} from "../interfaces/KeeperRegistryInterface1_2.sol";
```
Comment (infiloop2) [inline contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol:4]: Same question as canary wrapper.

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Added a couple of comments to discuss. 

We should figure out which of our interfaces we want to be versioned, and then any contract using that interface should also be versioned imo.

File: contracts/src/v0.8/KeeperRegistry1_2.sol (hunk 2 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/KeeperRegistry1_2.sol:16]: We will have a new UpkeepFormat (from V1 to both V1 & V2). Maybe we can just modify the existing UpkeepFormat enum sol file and not changing these.

File: contracts/src/v0.8/dev/CanaryUpkeep.sol (hunk 3 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/dev/CanaryUpkeep.sol:5]: Done

File: contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol (hunk 4 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol:4]: Done

**Comment by FelixFan1992** [comment]
> 

I think we can only need to version the `KeeperRegistryInterface.sol`.
for UpkeepFormat, we can keep appending to it and we don't need to change the `MigratableKeeperRegistryInterface` and `UpkeepTranscoderInterface`.

File: contracts/src/v0.8/KeeperRegistry1_2.sol (hunk 2 above)
Comment (infiloop2) [reply contracts/src/v0.8/KeeperRegistry1_2.sol]: I think it's fine to keep a single upkeep format file across different versions as we just append to enum

**Comment by infiloop2** [comment]
I think UpkeepTranscoder.sol we can also keep a single file since that is meant to work across versions for migration

**Comment by FelixFan1992** [comment]
> I think UpkeepTranscoder.sol we can also keep a single file since that is meant to work across versions for migration

Done

**Review by infiloop2 — APPROVED** [review]
lgtm!

**Pull request merged**